features/netsocket/LoRaWANBase.h

```diff
@@ -0,0 +1,222 @@
+     *                    things while being in a different region.
+     * @return            LORA_MAC_STATUS_OK if everything goes well, otherwise
+     *                    a negative error code.
+     */
+    virtual lora_mac_status_t set_datarate(uint8_t data_rate) = 0;
+
+    /** Enables adaptive data rate (ADR)
+     *
+     * Underlying LoRaPHY and LoRaMac layers handle the data rate automatically
+     * for the user based upon radio conditions (network congestion).
+     */
+    virtual lora_mac_status_t enable_adaptive_datarate() = 0;
+
+    /** Disables adaptive data rate
+     *
+     * When adaptive data rate (ADR) is disabled, user can either set a certain
+     * data rate or the Mac layer will choose a default value.
+     */
+    virtual lora_mac_status_t disable_adaptive_datarate() = 0;
+
+    /** Sets up retry counter for confirmed messages
+     *
+     * Valid only for confirmed messages.
+     *
+     * Number of trials to transmit the frame, if the LoRaMAC layer did not
+     * receive an acknowledgment. The MAC performs a data-rate adaptation,
+     * according to the LoRaWAN Specification V1.0.2, chapter 18.4, according
+     * to the table on page 64.
+     *
+     * Note, that if NbTrials is set to 1 or 2, the MAC will not decrease
+     * the datarate, in case the LoRaMAC layer did not receive an acknowledgment.
+     *
+     * @param count     number of retries for confirmed messages
+     *
+     * @return      LORA_MAC_STATUS_OK or a negative error code
+     */
+    virtual lora_mac_status_t set_confirmed_msg_retries(uint8_t count) = 0;
+
+    /** Sets channel plan
+     *
+     * @param channel_plan  The defined channel plans to be set.
+     * @return              0 on success, a negative error code on failure.
+     */
+    virtual lora_mac_status_t set_channel_plan(const lora_channelplan_t &channel_plan) = 0;
+
+    /** Gets the current channel plan.
+     *
+     * @param  channel_plan     The current channel information.
+     * @return                  0 on success, a negative error code on failure.
+     */
+    virtual lora_mac_status_t get_channel_plan(lora_channelplan_t &channel_plan) = 0;
+
+    /** Removes currently active channel plan
+     *
+     * Default channels (channels where Base Stations are listening) are not
+     * allowed to be removed. So when a plan is abolished, only non-default
+     * channels are removed.
+     *
+     * @return                  0 on success, negative error code on failure
+     */
+    virtual lora_mac_status_t remove_channel_plan() = 0;
+
+    /** Removes a given single channel
+     *
+     * Default channels (channels where Base Stations are listening) are not
+     * allowed to be removed.
+     *
+     * @param    index          The channel index
+     * @return                  0 on success, negative error code on failure
+     */
+    virtual lora_mac_status_t remove_channel(uint8_t index) = 0;
```

Comment:
Shouldn't this be instead of 0, the real status code, I can see statuses defined for `lora_mac_status_t `. Initially I thougth this is just an integer